File: 5-Problem-A-Day/66. Plus One.java
class Solution {
    public int[] plusOne(int[] digits) {
        int[] ans = new int[digits.length + 1];
        int carry = 1;
        for (int i = digits.length - 1; i > -1; i--) {
            int tmp = digits[i] + carry;
            ans[i + 1] = tmp % 10;
            digits[i] = tmp % 10;
            carry = tmp / 10;
        }
        if (carry != 0) {
            ans[0] = carry;
            return ans;
        } else
            return digits;
    }
}

File: 2024/66. Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry = 1;
        vector<int> res;
        for (int i = digits.size() - 1; i >= 0; i--) {
            int sum = digits[i] + carry;
            res.push_back(sum % 10);
            carry = sum / 10;
        }
        if (carry > 0) {
            res.push_back(carry);
        }
        reverse(res.begin(), res.end());
        return res;
    }
};